nca/FWRules/ConnectivityGraph.py

```diff
@@ -51,6 +68,42 @@ def add_edges(self, connections):
         """
         self.connections_to_peers.update(connections)
 
+    def add_edges_from_cube_dict(self, conn_cube, ip_blocks_mask):
+        """
+        Add edges to the graph according to the give cube
+        :param ConnectivityCube conn_cube: the given cube
+        :param IpBlock ip_blocks_mask:  IpBlock containing all allowed ip values,
+         whereas all other values should be filtered out in the output
+        """
+        src_peers = conn_cube["src_peers"]
+        conn_cube.unset_dim("src_peers")
+        dst_peers = conn_cube["dst_peers"]
+        conn_cube.unset_dim("dst_peers")
+        if IpBlock.get_all_ips_block() != ip_blocks_mask:
+            src_peers.filter_ipv6_blocks(ip_blocks_mask)
+            dst_peers.filter_ipv6_blocks(ip_blocks_mask)
+        protocols = conn_cube["protocols"]
+        conn_cube.unset_dim("protocols")
+
+        if protocols.is_whole_range() and not conn_cube.has_active_dim():
+            conns = ConnectionSet(True)
+        else:
+            conns = ConnectionSet()
+            protocol_names = ProtocolSet.get_protocol_names_from_interval_set(protocols) if protocols else ['TCP']
```

Comment:
same question here (as in the `ConnectionSet`)

Reply:
Added assertion (as in the `ConnectionSet`)